scripts/before_install_firestore_cmake.sh

```diff
@@ -0,0 +1,22 @@
+# Copyright 2018 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+bundle install
```

Comment:
The other travis-aware scripts in here basically switch on $PROJECT and $METHOD. I think you should do this too.

Make two scripts:
  * `install_prereqs.sh` that switches on `$METHOD`. You can also use this in the `Firebase` builds
  * `pod_install.sh` that switch on `$PROJECT` and `$METHOD` to do the right thing.

Then every `test` stage is basically the same:

```
    - stage: test
      env:
        - PROJECT=Firestore PLATFORM=macOS METHOD=cmake
      before_install:
        - ./scripts/if_changed.sh ./scripts/install_prereqs.sh
        - ./scripts/if_changed.sh ./scripts/pod_install.sh
      script:
        - ./scripts/if_changed.sh ./scripts/build.sh $PROJECT $PLATFORM $METHOD
```
Bonus points for making these scripts natively reuse `scripts/if_changed.sh` so that the per-stage stanzas have even less in them.

Reply:
Thanks for the ideas about using `if_changed` and about `pod_install`. By the way, is there any particular reason we're using `before_install` and not just `install`?

Reply:
Re `before_install` vs `install` I don't really know, though Travis's own docs suggest that we should install things like build software in `before_install` and things like our dependent software (pod install) via `install`.

I don't know if there are any consequences one way or another.